pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -72,10 +72,30 @@ func (pg *pgcontroller) addReplicaSet(obj interface{}) {
 
 	if *rs.Spec.Replicas == 0 {
 		pgName := batchv1alpha1.PodgroupNamePrefix + string(rs.UID)
+		klog.V(4).Infof("Delete podgroup %s for replicaset %s/%s spec.replicas == 0",
+			pgName, rs.Namespace, rs.Name)
 		err := pg.vcClient.SchedulingV1beta1().PodGroups(rs.Namespace).Delete(context.TODO(), pgName, metav1.DeleteOptions{})
 		if err != nil && !apierrors.IsNotFound(err) {
 			klog.Errorf("Failed to delete PodGroup <%s/%s>: %v", rs.Namespace, pgName, err)
 		}
+	} else if *rs.Spec.Replicas > 0 {
+		selector := metav1.LabelSelector{MatchLabels: rs.Spec.Selector.MatchLabels}
+		podList, err := pg.kubeClient.CoreV1().Pods(rs.Namespace).List(context.TODO(),
+			metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(&selector)})
+		if err != nil {
+			klog.Errorf("Failed to list pods for ReplicaSet <%s/%s>: %v", rs.Namespace, rs.Name, err)
+			return
+		}
+		if podList != nil {
+			for _, pod := range podList.Items {
+				klog.V(4).Infof("Try to create podgroup for pod %s/%s", pod.Namespace, pod.Name)
+				err := pg.createNormalPodPGIfNotExist(&pod)
```

Comment:
How about just check whether rs associated pg exists first, only create pg when it doesn't exist to avoid unnecessary API request, cause it's a rare case here.

Reply:
createNormalPodPGIfNotExist will checks if the podgroup exists.

![image](https://github.com/volcano-sh/volcano/assets/58628346/e8448a78-1260-4bf8-a5cb-40e62c574ac0)

Reply:
Ok.